mm2src/coins/eth/v2_activation.rs

```diff
@@ -916,3 +921,38 @@ fn compress_public_key(uncompressed: H520) -> MmResult<H264, EthActivationV2Erro
     let compressed = public_key.serialize();
     Ok(H264::from(compressed))
 }
+
+#[cfg(not(target_arch = "wasm32"))]
+async fn run_db_migration_for_new_eth_pubkey(ctx: &MmArc, db_id: String) -> MmResult<(), EthActivationV2Error> {
+    info!("Public key hash: {db_id:?}");
+
+    let mut db_migration_sender = ctx
+        .db_migration_watcher
+        .as_option()
+        .expect("Db migration watcher isn't intialized yet!")
+        .get_sender();
+    db_migration_sender
+        .send(db_id)
+        .await
+        .map_to_mm(|err| EthActivationV2Error::InternalError(err.to_string()))?;
+
+    Ok(())
+}
```

Comment:
This block of code is repeated for eth, tendermint and utxo. You should have only one implementation of it with a custom error that gets converted to different activation errors.

Reply:
refactored. thanks